### PR TITLE
Scale test cases for pvc expansion.

### PR DESCRIPTION
**Pull request by tiffanyn108**
Signed-off-by: Tiffany Nguyen <tunguyen@redhat.com>

### CODE REVIEW

**Comment by ramkiperiy** [comment]
@tiffanyn108 PR looks good, do you have test run in local with less scale pod count, if so please do add that logs

File: ocs_ci/ocs/scale_lib.py [hunk 1]
```diff
@@ -92,7 +93,10 @@ def create_multi_pvc_pod(
         """
         rbd_sc = helpers.default_storage_class(constants.CEPHBLOCKPOOL)
         cephfs_sc = helpers.default_storage_class(constants.CEPHFILESYSTEM)
-        pvc_size = f"{random.randrange(15, 105, 5)}Gi"
+        if not pvc_size:
+            pvc_size = f"{random.randrange(15, 105, 5)}Gi"
+        else:
+            pvc_size = pvc_size
```
Comment (jilju) [inline ocs_ci/ocs/scale_lib.py:99]:
```suggestion
        pvc_size = pvc_size or f"{random.randrange(15, 105, 5)}Gi"
```

File: ocs_ci/ocs/scale_lib.py [hunk 2]
```diff
@@ -74,7 +74,8 @@ def create_and_set_namespace(self):
             self.sa_name = None
 
     def create_multi_pvc_pod(
-        self, pods_per_iter=5, io_runtime=3600, start_io=False
+        self, pods_per_iter=5, io_runtime=3600, start_io=False,
+        pvc_size=None
```
Comment (jilju) [inline ocs_ci/ocs/scale_lib.py:78]: Add 'pvc_size' in docstring

File: ocs_ci/ocs/scale_lib.py [hunk 3]
```diff
@@ -168,7 +173,8 @@ def create_multi_pvc_pod(
         return pod_objs, pvc_objs
 
     def create_scale_pods(
-        self, scale_count=1500, pods_per_iter=5, io_runtime=None, start_io=None
+        self, scale_count=1500, pods_per_iter=5, io_runtime=None,
+        pvc_size=None, start_io=None
```
Comment (jilju) [inline ocs_ci/ocs/scale_lib.py:177]: Please make sure the signature change is not affecting existing usages.

Comment (jilju) [inline ocs_ci/ocs/scale_lib.py:177]: Add 'pvc_size' in docstring

File: ocs_ci/ocs/scale_lib.py [hunk 4]
```diff
@@ -242,6 +248,15 @@ def create_scale_pods(
                         "Scaling PVC+POD failed analyze setup and log for more details"
                     )
 
+    def pvc_expansion(self, pvc_new_size):
+        """
+        Function to expand PVC size and verify the new size is reflected.
+        """
+        logging.info(f"PVC size is expanding to {pvc_new_size}")
+        for pvc_oj in self.pvc_obj:
```
Comment (keesturam) [inline ocs_ci/ocs/scale_lib.py:256]: did you mean `pvc_obj`?

File: tests/e2e/scale/test_scale_pvc_expand.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    E2ETest, scale, ignore_leftovers, skipif_ocs_version
+)
+from ocs_ci.ocs.scale_lib import FioPodScale
+from ocs_ci.utility import utils
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='session')
+def resize_pvc(request):
```
Comment (keesturam) [inline tests/e2e/scale/test_scale_pvc_expand.py:14]: There seems to be already a function with the same name `resize_pvc` in pvc.py. do you want to use a different name here?

File: ocs_ci/ocs/scale_lib.py (hunk 1 above)
Comment (tiffanyn108) [reply ocs_ci/ocs/scale_lib.py:99]: Ack.

File: ocs_ci/ocs/scale_lib.py (hunk 2 above)
Comment (tiffanyn108) [reply ocs_ci/ocs/scale_lib.py:78]: Acked.

File: ocs_ci/ocs/scale_lib.py (hunk 3 above)
Comment (tiffanyn108) [reply ocs_ci/ocs/scale_lib.py:177]: It will not change the existing usages.  Added doc string for create_scale_pods() as well.

Comment (tiffanyn108) [reply ocs_ci/ocs/scale_lib.py:177]: Acked.

File: ocs_ci/ocs/scale_lib.py (hunk 4 above)
Comment (tiffanyn108) [reply ocs_ci/ocs/scale_lib.py:256]: It is pvc in pvc_obj.  I'm changing to pvc_object instead.

File: tests/e2e/scale/test_scale_pvc_expand.py (hunk 5 above)
Comment (tiffanyn108) [reply tests/e2e/scale/test_scale_pvc_expand.py:14]: I'm calling the same function resize_pvc() in pvc.py thus name should be the same.

**Comment by tiffanyn108** [comment]
Local run with 100 pvcs and it passed in both case with IO and without IO.
https://paste.centos.org/view/829c8776

**Pull request merged**